source/dynamical_systems/include/dynamical_systems/Linear.hpp

```diff
@@ -37,29 +39,34 @@ class Linear : public DynamicalSystem<S> {
    * @param state the input state
    * @return the output state
    */
-  S compute_dynamics(const S& state) const;
+  S compute_dynamics(const S& state) const override;
 
 public:
+  /**
+   * @brief Empty constructor
+   */
+  Linear();
+
   /**
    * @brief Constructor with specified attractor and iso gain
    * @param attractor the attractor of the linear system
    * @param iso_gain the iso gain of the system
    */
-  explicit Linear(const S& attractor, double iso_gain = 1.0);
+  Linear(const S& attractor, double iso_gain = 1.0);
```

Comment:
Why did you remove the `explicit` for those? I think for the empty constructor it makes sense but here they should be kept

Reply:
@eeberhard advised me to remove `explicit` for the Jacobian constructors too, which have more than one argument. I assumed this is applicable here too

Reply:
It's a little different here because this can be treated as a single argument constructor (because of the default value). So, explicit does have a purpose - if you keep it, then the passed attractor type _must_ be the same as the class template type. Without explicit, you could do `Linear<CartesianState>(CartesianPose::Identity(...))`. So the question is more about which behaviour you want. The rule of thumb is to remove `explicit` from constructors that have 0 or >1 argument (excluding default arguments), and to think about the desired behaviour single-argument cases.

Reply:
I am pretty sure I was using `Linear<CartesianState>(CartesianPose)` even with explicit. But probably because a `CartesianPose` is also a `CartesianState`. To be honest I never really know what are the actual implications for it but I am happy to have some rules that we always apply